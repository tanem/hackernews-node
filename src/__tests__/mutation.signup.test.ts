import Chance from 'chance'
import path from 'path'
import request from 'supertest'

const chance = Chance(path.basename(__filename))

let httpServerUrl: typeof global.__HTTP_SERVER_URL__

beforeAll(() => {
  httpServerUrl = global.__HTTP_SERVER_URL__
})

const name = chance.first()
const email = chance.email()
const password = chance.word()

test('signup', async () => {
  const { body, status } = await request(httpServerUrl)
    .post('/')
    .send({
      query: `
      mutation {
        signup(
          name: "${name}"
          email: "${email}"
          password: "${password}"
        ) {
          token
          user {
            id
          }
        }
      }
    `,
    })

  expect(status).toBe(200)
  expect(body).toMatchSnapshot({
    data: {
      signup: {
        token: expect.any(String),
        user: {
          id: expect.any(String),
        },
      },
    },
  })
})
