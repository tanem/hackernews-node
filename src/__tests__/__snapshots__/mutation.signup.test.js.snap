// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signup 1`] = `
Object {
  "data": Object {
    "signup": Object {
      "token": Any<String>,
      "user": Object {
        "id": Any<String>,
      },
    },
  },
}
`;
